jenkins-scripts/docker/lib/docker_generate_dockerfile.bash

```diff
@@ -40,48 +40,50 @@ fi
 
 export APT_PARAMS=
 
-GZDEV_DIR=${WORKSPACE}/gzdev
+GZDEV_DIR=/root/gzdev
 GZDEV_BRANCH=${GZDEV_BRANCH:-master}
 if python3 ${SCRIPT_DIR}/../tools/detect_ci_matching_branch.py "${ghprbSourceBranch}"; then
   GZDEV_TRY_BRANCH=$ghprbSourceBranch
 fi
- 
+
 KEYSERVER="keyserver.ubuntu.com"
 if [[ "${DISTRO}" == 'bionic' || "${DISTRO}" == 'focal' ]]; then
   KEYSERVER="hkps://pgp.surf.nl"
 fi
 
 dockerfile_install_gzdev_repos()
 {
-cat >> Dockerfile << DELIM_OSRF_REPO_GIT
-RUN rm -fr ${GZDEV_DIR}
-RUN git clone https://github.com/gazebo-tooling/gzdev -b ${GZDEV_BRANCH} ${GZDEV_DIR}
-RUN if [ -n $GZDEV_TRY_BRANCH ]; then \
```

Comment:
To support also docker caching the `GZDEV_TRY_BRANCH` this check has been taken out of the Dockerfile and a new ADD to check the GZDEV_TRY_BRANCH was added in adding to the two git commands.